Mathlib/Analysis/Normed/Affine/Simplex.lean

```diff
@@ -0,0 +1,183 @@
+/-
+Copyright (c) 2025 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import Mathlib.Analysis.Normed.Group.AddTorsor
+import Mathlib.LinearAlgebra.AffineSpace.Independent
+
+/-!
+# Simplices in torsors over normed spaces.
+
+This file defines properties of simplices in a `NormedAddTorsor`.
+
+## Main definitions
+
+* `Affine.Simplex.Scalene`
+* `Affine.Simplex.Equilateral`
+* `Affine.Simplex.Regular`
+
+-/
+
+
+namespace Affine
+
+open Function
+
+variable {R V P : Type*} [Ring R] [SeminormedAddCommGroup V] [PseudoMetricSpace P] [Module R V]
+variable [NormedAddTorsor V P]
+
+namespace Simplex
+
+variable {m n : ℕ}
+
+/-- A simplex is scalene if all the edge lengths are different. -/
+def Scalene (s : Simplex R P n) : Prop :=
+  Injective fun i : {x : Fin (n + 1) × Fin (n + 1) // x.1 < x.2} ↦
+    dist (s.points i.val.1) (s.points i.val.2)
+
+lemma Scalene.dist_ne {s : Simplex R P n} (hs : s.Scalene) {i₁ i₂ i₃ i₄ : Fin (n + 1)}
+    (h₁₂ : i₁ ≠ i₂) (h₃₄ : i₃ ≠ i₄) (h₁₂₃₄ : i₁ ≠ i₃ ∨ i₂ ≠ i₄) (h₁₂₄₃ : i₁ ≠ i₄ ∨ i₂ ≠ i₃) :
```

Comment:
Just wondering, would this be easier to apply if the latter two assumptions were phrased as `not (i₁ = i₃ \and i₂ = i₄)`. Because then, when proving the assumption, you can immediately `rintro \<rfl, rfl\>`, and work towards a contradiction.

Reply:
I've made that change.